homework-submission/react-hyf/client/src/components/week3-homework/week3.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react';
+import User from './user/user';
+import Spinner from './spinner/spinner';
+import './week3.css';
+
+class Week3 extends Component {
+  state = { users: [], loading: true, show: true };
+
+  componentDidMount() {
+    this.setFocus();
+    this.setState({ show: true });
+  }
+
+  handleKeyPress = e => {
+    if (e.key === ' ' || e.key === 'Spacebar') {
+      fetch(' https://uinames.com/api/?amount=10')
+        .then(res => res.json())
+        .then(result => {
+          const amount = result.slice(1, 2);
+          this.setState({ users: amount, loading: false, show: false });
+        });
+    }
+  };
+
```

Comment:
Because you are handling the `fetch` inside the handleKeyPress the page won't fetch any data until you press a key, you could separate the function and also call that function in `componentDidMount`. (`this.fetchData()`)